### PR TITLE
Add doc for snapshot validation webhook

### PR DESCRIPTION
**Pull request by xing-yang**

### CODE REVIEW

File: book/src/SUMMARY.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
     - [Kubernetes Changelog](kubernetes-changelog.md)
     - [Kubernetes Cluster Controllers](kubernetes-cluster-controllers.md)
         - [Snapshot Controller](snapshot-controller.md)
+	- [Snapshot Validation Webhook](snapshot-validation-webhook.md)
```
Comment (msau42) [inline book/src/SUMMARY.md:8]: spacing doesn't seem correct

File: book/src/snapshot-restore-feature.md [hunk 2]
```diff
@@ -55,15 +55,21 @@ The snapshot controller is deployed by the Kubernetes distributions and is respo
 
 The CSI external-snapshotter sidecar watches Kubernetes VolumeSnapshotContent CRD objects and triggers CreateSnapshot/DeleteSnapshot against a CSI endpoint.
 
+### Snapshot Validation Webhook
+
+There is a new validating webhook server which provides tightened validation on snapshot objects. This SHOULD be installed by all users of this feature. See [Snapshot Validation Webhook](snapshot-validation-webhook.md) for more details on how to use the webhook.
```
Comment (msau42) [inline book/src/snapshot-restore-feature.md:60]: clarify here that this should be installed by the Kubernetes distros along with the snapshot-controller; not end users

File: book/src/snapshot-validation-webhook.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Snapshot Validation Webhook
+
+## Status and Releases
+
+**Git Repository:** [https://github.com/kubernetes-csi/external-snapshotter](https://github.com/kubernetes-csi/external-snapshotter)
+
+**Status:** v3.0.1
```
Comment (msau42) [inline book/src/snapshot-validation-webhook.md:7]: note this is beta

File: book/src/snapshot-validation-webhook.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+### Snapshot Validation Webhook
+
+There is a new validating webhook server which provides tightened validation on snapshot objects. This SHOULD be installed by all users of the snapshot feature.
```
Comment (msau42) [inline book/src/snapshot-validation-webhook.md:11]: clarify here it should be the responsibility of k8s distros

**Review by msau42 — COMMENTED** [review]
As a followup, can we also add a section under snapshot apis talking about the client libaries? Also, do we have any generated openapi docs we can host for the crds?

File: book/src/SUMMARY.md (hunk 1 above)
Comment (xing-yang) [reply book/src/SUMMARY.md]: done

File: book/src/snapshot-restore-feature.md (hunk 2 above)
Comment (xing-yang) [reply book/src/snapshot-restore-feature.md]: done

File: book/src/snapshot-validation-webhook.md (hunk 3 above)
Comment (xing-yang) [reply book/src/snapshot-validation-webhook.md]: done

File: book/src/snapshot-validation-webhook.md (hunk 4 above)
Comment (xing-yang) [reply book/src/snapshot-validation-webhook.md]: done

**Review by xing-yang — COMMENTED** [review]
@msau42 Addressed your comments.  PTAL.

**Comment by xing-yang** [comment]
> As a followup, can we also add a section under snapshot apis talking about the client libaries? Also, do we have any generated openapi docs we can host for the crds?

Sure, I'll add a section for the client lib in a followup PR.
For the generated openapi docs, @humblec has a PR in external-snapshotter repo, but I think csi docs is a better place for that.

**Pull request merged**